### ISSUE TITLE
Sanitize Log Group and Stream Names

### ISSUE THREAD

**Issue by PettitWesley** [aws/amazon-cloudwatch-logs-for-fluent-bit#103]
With the new templating feature, in ECS FireLens you might be tempted to do something like:

```
log_stream_name    something-$(ecs_task_arn)
```

But this leads to an error:

```
INFO[0008] [cloudwatch 0] Log group test already exists
ERRO[0008] [cloudwatch 0] InvalidParameterException: 1 validation error detected: Value 'test-arn:aws:ecs:ap-south-1:144718711470:task/737d73bf-8c6e-44f1-aa86-7b3ae3922011' at 'logStreamName' failed to satisfy constraint: Member must satisfy regular expression pattern: [^:*]*
```

Because the colon character makes it an invalid stream name. 

Ideally, the plugin should just strip colons from the names in this case.

**Comment by PettitWesley** [comment]
CC @davidnewhall

**Comment by davidnewhall** [comment]
Slicing the wholly unknown data in that json payload seems.. less than ideal. Every piece we add makes the app slower.

Deleting colons or replacing them is probably OK if this really seems like a use-case folks will need. 

How would you like it handled? Can get a PR for that pretty easy (probably a 1-liner).

**Comment by PettitWesley** [comment]
@davidnewhall Dots are commonly used a separator in the fluent world, I'd just do a `strings.ReplaceAll(name, ":", ".")` or something.

**Comment by davidnewhall** [comment]
This may be closed. Colons and other bad characters are now replaced with period/dots. Group and stream names are also truncated to their max size of 512 characters.

**Comment by PettitWesley** [comment]
Thanks!

Released in 2.9.0
